### PR TITLE
Add timesheet

### PR DESCRIPTION
**Pull request by ivuorio**

### CODE REVIEW

File: src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import three.consulting.epoc.entity.Timesheet
+import java.time.LocalDateTime
+import javax.validation.constraints.NotBlank
+
+data class TimesheetDTO(
+    val id: Long? = null,
+    @field:NotBlank val name: String,
+    val description: String? = null,
+    val allocation: Double,
```
Comment (ivuorio) [inline src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt:11]: What is your take on the format for allocation

Comment (LeadingMoominExpert) [reply src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt:11]: I'd prefer 0-100. I don't know if @robertsiipola has a prefence what he'd like in the frontend side. Anyhoo, if we use 0-100 this should be int and we can validate the field with
```kotlin
@field:Min(0) @field:Max(100) val allocation: Int
```

File: src/main/resources/db/migration/V4__Create_Timesheet.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+);
+
+ALTER TABLE timesheet ADD CONSTRAINT FK_TIMESHEET_ON_EMPLOYEE FOREIGN KEY (employee_id) REFERENCES employee (id);
+
+ALTER TABLE timesheet ADD CONSTRAINT FK_TIMESHEET_ON_PROJECT FOREIGN KEY (project_id) REFERENCES project (id);
+
+INSERT INTO timesheet(project_id,employee_id,name,description,allocation,created,updated)
+VALUES
+    (1,1,'test','testing',100,CURRENT_TIMESTAMP, CURRENT_TIMESTAMP ),
+    (1,1,'test2','testing delete',100,CURRENT_TIMESTAMP, CURRENT_TIMESTAMP)
```
Comment (LeadingMoominExpert) [inline src/main/resources/db/migration/V4__Create_Timesheet.sql:20]: Not sure if it's needed but i'd add semicolon after the last sql statement too just in case.

File: src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt (hunk 1 above)
Comment (LeadingMoominExpert) [reply src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt:11]: And if you add the Min/Max validation please add a couple tests to see the request fails when the value is outside those boundaries.

**Review by LeadingMoominExpert — CHANGES_REQUESTED** [review]

File: src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt (hunk 1 above)
Comment (robertsiipola) [reply src/main/kotlin/three/consulting/epoc/dto/TimesheetDTO.kt:11]: 0-100 is fine with me 👍🏼

**Review by LeadingMoominExpert — APPROVED** [review]
